File: Java/Medium/WrongAnswer/smallest-string-with-a-given-numeric-value_8.java
class Solution {
    public String getSmallestString(int n, int k) {
        
        char[] s = new char[n];
        Arrays.fill(s, 'a');
        int i = n - 1;
        while (k > n) {
            s[i] = (char) ('a' + Math.min(25, k - n));
            k -= s[i] - 'a' + 1;
            i--;
        }
        s[i] = (char) ('a' + k - 1);
        return new String(s);
    }
}

File: Rust/Medium/WrongAnswer/smallest-string-with-a-given-numeric-value_8.rs
impl Solution {
    pub fn get_smallest_string(n: i32, k: i32) -> String {
        
        let mut res = String::new();
        let mut k = k;
        for i in 0..n-1 {
            let mut c = k - n + i + 1;
            if c > 26 {
                c = 26;
            }
            res.push((c + 96) as u8 as char);
            k -= c;
        }
        res.push((k + 96) as u8 as char);
        res
    }
}